framework/src/ics/InitialConditionWarehouse.C

```diff
@@ -43,43 +45,44 @@ InitialConditionWarehouse::addObject(std::shared_ptr<InitialConditionBase> objec
       mooseError("You are trying to set a boundary restricted variable on non-nodal variable. That "
                  "is not allowed.");
 
-    std::map<std::string, std::set<BoundaryID>>::const_iterator iter =
-        _boundary_ics[tid].find(var.name());
+    std::map<std::tuple<std::string, int>, std::set<BoundaryID>>::const_iterator iter =
```

Comment:
```suggestion
    auto iter =
```

let's use `auto` where we can